### ISSUE TITLE
ModelSelect2 widget breaks when admin view has inlines

### ISSUE THREAD

**Issue by matthew-nm** [yourlabs/django-autocomplete-light#1110]
Django: 2.2.1
DAL: 3.3.5

The title basically says it all, but I will be happy to create a fork of the project with this issue spelled out more if there is no known obvious fix (I can't believe I'm the first person to run into this issue, but I suppose it's possible!).

The ModelSelect2 widget works fine elsewhere in my site, and it even works in the case I'm having problems with if I remove the inlines. The inlines have nothing to do with the widget, but somehow they don't play nice together.

**Comment by bpepple** [comment]
Confirming this bug. Just ran into it on a project that uses ModelSelect2 widget inline on a form.

**Comment by matthew-nm** [comment]
@bpepple Is your ModelSelect2 widget used within the inline? That sounds like a slightly different scenario than mine (widget is in main area of form, but when inlines are also present, widget breaks). Could still be the same underlying issue.

**Comment by nameisnot** [comment]
I am getting a similar issue but with ModelSelect2 in inlines as well. I am using dynamic formset with extra set a value more than 0. When the form loads the widgets work as expected. But when a new line of forms is added the line before (the added new one) loses its DAL features, with only the selected value visible/available.

**Comment by jpic** [comment]
It's working for me in the example apps in the test project:

![2019-09-16-130512_1347x509_scrot](https://user-images.githubusercontent.com/94636/64953329-dc025800-d882-11e9-8a2d-25c4865cd7c2.png)

Is there any other way to reproduce the issue ?

**Comment by nameisnot** [comment]
DAL is working as expected in **admin** (as you have shown above). But I am having trouble in the **frontend**. Following images (from the inline part of the formset) display as to how DAL is failing:

![1 INITIAL SCRN](https://user-images.githubusercontent.com/40018225/65054929-fb38dc80-d98b-11e9-9dcf-9c39190f72f3.png)

![2 1ST FORMSET FILLED](https://user-images.githubusercontent.com/40018225/65054969-0c81e900-d98c-11e9-8827-ee7119142e80.png)

![3 1ST FORMSET FILLED N NEW LINE ADDED](https://user-images.githubusercontent.com/40018225/65054979-1277ca00-d98c-11e9-81c7-3d942db4c215.png)

![4 DAL IS DEAD NOW](https://user-images.githubusercontent.com/40018225/65055001-1ad00500-d98c-11e9-8219-e2d250cf14a1.png)

What could be wrong?

**Comment by nameisnot** [comment]
What is the implication of the [assertion](https://github.com/yourlabs/django-autocomplete-light/issues/1082#issuecomment-458412758)?

How to implement  the suggestion (I have the checked autocomplete.init.js and find that the code line :

`$(e.target).find('[data-autocomplete-light-function=select2]:not([id*="__prefix__"])').each(initialize);
`

is still (in) there. Or am I missing something here? What (or how) do I implement what has been suggested by @dwasyl?

**A side note:** I realised that the above images without explanation may have lost its intended purpose which is why I am going to add the circumstances under which I get these.

**Comment by nameisnot** [comment]
In order for you to get a ringside view, I am reproducing the relevant parts of the project I am using the autocomplete on (in case I am missing out on something, please do let me know):

**models.py**

```
class PurchOrder(models.Model):
    PO_Number = models.AutoField(primary_key=True, verbose_name='PO Number')
    po_date = models.DateTimeField(default=datetime.now, null=True, verbose_name='PO date')
    po_generating_plant = models.ForeignKey(Plant, related_name='po_plant', on_delete=models.CASCADE, verbose_name='Orig.Plant')
    po_supplier = models.ForeignKey(Supplier, on_delete=models.CASCADE, null=True, verbose_name='Vendor')

    def __str__(self):
        return '{0}'.format(self.PO_Number)


class PurchOrderItem(models.Model):
    po_item_num = models.AutoField(primary_key=True, verbose_name='Line Item Number')
    hdr_po_num = models.ForeignKey(PurchOrder, on_delete=models.CASCADE, null=True, verbose_name='PO Number')
    po_itm_matl_type = models.ForeignKey('MaterialTyp', on_delete=models.SET_NULL, null=True, verbose_name='Matl Type')
    po_itm_itemcatgroup = models.ForeignKey(ItmCategoryGroupMaterialTypAssginment, on_delete=models.CASCADE, null=True, verbose_name='ItmCat Group')
    po_itm_material = models.ForeignKey('Material', on_delete=models.CASCADE, verbose_name='Material')
    po_itm_quantity = models.PositiveSmallIntegerField(null=True, default=0, verbose_name='Quantity')
    po_itm_quant_uom = models.ForeignKey(Unit_of_Measure, on_delete=models.CASCADE, null=True, verbose_name='UoM')
    po_itm_price = MoneyField(max_digits=9, decimal_places=2, null=True, default_currency='INR', verbose_name='Item Price')
    po_item_received = models.BooleanField(default=False, verbose_name='Recvd')
    
    def __str__(self):
        return '{0}'.format(self.po_item_num)
```

**forms.py**

```
class PurchOrderItemForm(forms.ModelForm):
    class Meta:
        model = PurchOrderItem
        fields = ('po_item_num', 'po_itm_matl_type', 'po_itm_itemcatgroup', 'po_itm_material', 'po_itm_quantity', 'po_itm_quant_uom', 'po_itm_price', 'po_itm_supply_by_date', 'po_itm_split_allowed')
        widgets = {
            'po_itm_matl_type': forms.Select(attrs={"onChange":'change_matltype()'}),
            'po_itm_supply_by_date': DatePickerInput(),
            'po_itm_supply_by_date': DatePickerInput(attrs={'style': 'width:180px'}),
            'po_itm_quantity': forms.NumberInput(attrs={'style': 'width:80px'}),
            'po_itm_matl_type': autocomplete.ModelSelect2(url='materialtyp_autocomplete',
                                                          attrs={'data-placeholder': 'Click here...',
                                                                 'data-minimum-input-length': 2}),
            'po_itm_itemcatgroup': autocomplete.ModelSelect2(url='itemcategorygroup_autocomplete',
                                                             forward=['po_itm_matl_type'],
                                                             attrs={'data-placeholder': 'Click here...', 'data-minimum-input-length': 3}),
            }
        exclude = ()


PurchOrderItemChangeFormset = inlineformset_factory(
    PurchOrder,
    PurchOrderItem,
    form=PurchOrderItemForm, can_delete=True, min_num=1, validate_min=True, max_num=100,
    extra=1)
```

**views.py**

```
class TestPurchOrderCreate(ModelFormWidgetMixin, CreateView):
    template_name = "matmovt/purchorder_create_with_lineitems.html" # OLD VALUE: purchorder_update_form.html"
    model = PurchOrder
    
    fields = ['po_doc_type', 'po_date', 'po_value', 'po_generating_plant', 'goods_receiving_plant', 'po_supplier', 'po_closed']

    widgets = {
        'po_date': DateTimePickerInput(),
        'po_doc_type': forms.TextInput(attrs={'readonly': 'readonly', 'style': 'width:60px; background:lightgrey'}),
        'po_value': forms.NumberInput(attrs={'readonly': 'readonly', 'style': 'width:100px; background:lightgrey'}),
        }

    success_url = reverse_lazy('purchorder_list')

    def clean_status(self):
        return self.instance.status
    
    def get_context_data(self, **kwargs):
        data = super(TestPurchOrderCreate, self).get_context_data(**kwargs)
        if self.request.POST:
            data['purchorderitem'] = PurchOrderItemChangeFormset(self.request.POST)
        else:
            data['purchorderitem'] = PurchOrderItemChangeFormset()
        return data

    def form_valid(self, form):
        context = self.get_context_data()
        purchorderitem = context['purchorderitem']
        with transaction.atomic():

            if purchorderitem.is_valid():
                self.object = form.save()                
                purchorderitem.instance = self.object
                purchorderitem.save()

            else:
                context.update({'purchorderitem': purchorderitem})
                return self.render_to_response(context)
        return super(TestPurchOrderCreate, self).form_valid(form)

    def get_success_url(self):
        return reverse_lazy('purchorder_list', kwargs={'pk': self.object.pk})

```

**Templates:**

**base.html**

```
<!DOCTYPE html>
<html lang="en">
{% load staticfiles %}
<head>
<title>MyLog</title>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1">

<script type="text/javascript" src="{{ STAIC_URL }} /static/jquery-3.4.1.js"> </script>

{% load bootstrap4 %}
{% bootstrap_css %}
{% bootstrap_javascript jquery='full' %}

{% block extrahead %}
	{# Embeds Extra Resources #}
{% endblock %}

<link rel="stylesheet" href="https://www.w3schools.com/w3css/4/w3.css">
<link rel="stylesheet" href="https://www.w3schools.com/lib/w3-theme-teal.css">

<!-- FOLL TWO JQRY BEING ADDED FOR "select2" implementation (select2.org) 15082019 -->

<link href="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.8/css/select2.min.css" rel="stylesheet" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.8/js/select2.min.js"></script>

<!-- SORTTABLE.JS ADDED 29/07/2019 TO SORT TABLES ON COLUMN CLICK -->
<script src="{% static "js/sorttable.js" %}" type="text/javascript"></script>


<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css"> 
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js"></script>


<script src="https://cdnjs.cloudflare.com/ajax/libs/malihu-custom-scrollbar-plugin/3.1.5/jquery.mCustomScrollbar.concat.min.js"></script>


<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/malihu-custom-scrollbar-plugin/3.1.5/jquery.mCustomScrollbar.min.css">


<!-- HERE I HAVE REMOVED SOME OF THE CSS -->


</head>


<header class="w3-container w3-theme w3-padding-12 w3-center">
  <h1 class="w3-xxxlarge w3-padding-16">Logistics</h1>
</header>

<nav class="navbar navbar-expand-sm sticky-top navbar-dark bg-secondary">
	<span class="navbar-toggler-icon leftmenutrigger"></span>
	<a class="navbar-brand" data-toggle="tooltip" title="MyLog Home" href="{% url "welcome" %}">
		<img src="https://coreui.io/docs/assets/brand/bootstrap-solid.svg" width="30" height="30" class="d-inline-block align-top" alt="">
		MyLog
	</a>

	<div class="collapse navbar-collapse" id="navbarSupportedContent">
		<ul class="navbar-nav mr-auto mt-2 mt-lg-0">
			<li class="nav-item active dropdown">
				<a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" title="HCM Main" aria-haspopup="true" aria-expanded="false">HCM <span class="sr-only">(current)</span></a>

				<div class="dropdown-menu" aria-labelledby="navbarDropdown">

					<a class="dropdown-item" href="{% url 'purchorder_new' %}">Purchase Order</a>
				</div>
			</li>
		</ul>
	</div>
</nav>

<!-- START OF MAIN PAGE AREA -->
<div class="col-sm-8">

	{% block content %}

	{% endblock %}


	{% block footer %}
<!-- TRIED THE FOOTER BLOCK IS FOR DJANGO AUTOCOMPLETE LIGHT  -->
<!-- BUT NOT MAKING ANY IMPROVEMENT IN PERFORMANCE / ALSO THE BS4 DATEPICKER IS RENDERED INACTIVE DUE TO THIS -->
	{% endblock %}


</body>

</html>

```

**purch_order.html**

```
{% extends "base.html" %}
<!-- BLOCKs "extrahead" BEING USED FOR THE FORM MEDIA - BOTH FOR BS4 DATEPICKER AND DAL -->
{% block extrahead %} 
{{ purchorderitem.media }} 
{% endblock %} 

{% load static %}


{% block content %}
<h4>Create Purchase Order</h4>

<hr>
<div class="container"> 
<div class="col-md-4">

<head>
<link rel="stylesheet" href="//code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css"> 
</head>

<form action="" method="POST" id="manufDomForm" data-cities-url="{% url 'ajax_load_cities' %}" novalidate> 
{% csrf_token %}

<div class="col-md-4 tablehdr-wrapper-scroll-y" style="font-family:'Courier New'">
<table id="poheader">
	{% for fieldY in form.visible_fields %}
		<tr>
		<th>{{ fieldY.label_tag }}</th>
			<td>
				{{ fieldY.errors }}
				{% if fieldY.name == "po_date" %}
					<div style="z-index: 999;" {{ fieldY }}</div>
				{% else %}
					{{ fieldY }}

				{% endif %}

				{% if fieldY.help_text %}
					{{ fieldY.help_text|safe }}
				{% endif %}


			</td>
		</tr>
	{% endfor %}
</table>
</div>
<br>

<!-- BELOW IS THE START OF MANAGEMENT_FORM FOR DISPLAYING FORMSETS -->
<div class="col-md-4 table-wrapper-scroll-y" style="font-family:'Courier New'">

		<table class="table table-bordered table-striped table-sm">
			{{ purchorderitem.management_form }}
			{% for form in purchorderitem.forms %}
				{% if forloop.first %}
					<thead id="polinehdr">
					<tr>
						{% for field in form.visible_fields %}
							<th>{{ field.label|capfirst }}</th>
						{% endfor %}
					</tr>
					</thead>
				{% endif %}
				<tr class="{% cycle row1 row2 %} formset_row"> 
					{% for field in form.visible_fields %}
						<td>
							{# Include the hidden fields in the form #}
							{% if forloop.first %}
								{% for hidden in form.hidden_fields %}
									{{ hidden }}
								{% endfor %}
							{% endif %}
							{{ field.errors.as_ul }}
							{% if field.name == "po_itm_supply_by_date" %}
								<div style="z-index: 99999;" {{ field }}</div>

							{% else %}
								{{ field }}
							{% endif %}
						</td>

					{% endfor %}
				</tr>
			{% endfor %}
		</table>
</div>

		<input type="submit" class="btn btn-info" value="Save"/> 
		<a href="{% url 'purchorder_list' %}" class="btn btn-info" role="button">Cancel</a>
<br>
</form>

</div>

{# SCRIPT FOR ADDING / REMOVING FORMSETS AT RUNTIME #}

    <script src="//ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
    <script src="{% static 'formset/jquery.formset.js' %}"></script>
    <script type="text/javascript">
        $('.formset_row').formset({
            addText: 'Add PO Line Items',
            deleteText: 'Remove',
            prefix: 'purchorderitem_set'
        });
    </script>


</div>
{% endblock %}

```


I have not included the **urls.py** (project and app) as well as the autocomplete views for brevity. As you would imagine I have tried many of the suggestions available relating to DAL including [this](https://django-autocomplete-light.readthedocs.io/en/master/tutorial.html#using-autocompletes-outside-the-admin) and [this](https://github.com/yourlabs/django-autocomplete-light/issues/1082#issuecomment-458412758) but have not succeeded so far.

**Comment by nameisnot** [comment]
@jpic  - I am sorry to have to bother you with this. Could you please guide me to a **minimal example project** where DAL has been used on **inlines** (outside of admin).

**PS.** As you might have noticed I am using **bootstrap datepicker plus** and taking the `{{ form.media }}` to the footer **renders the date widget inactive**.

**Comment by jpic** [comment]
Ok, but right now I got a lot of backlog, I can refer you to this article dissecting how it works in the admin so that you can reproduce : https://blog.yourlabs.org/post/20001556462/howto-javascript-popup-form-returning-value-for

**Comment by jpic** [comment]
Ok, but right now I got a lot of backlog, did you try importing the admin js and way of doing ? Last time I checked, they produced a hidden inline that would serve as template and be copied over after replacing the form prefix that is `__prefix__` for the template, which contains the inline number.
How it works in DAL is just by watching the DOMNodeInserted event, when it triggers it will re-initialize the JS, but it's got to insert a select that was *not* initialized because it's `__prefix__`: https://github.com/yourlabs/django-autocomplete-light/blob/master/src/dal/static/autocomplete_light/autocomplete.init.js#L151-L155
As such, initialization of newly insert copy on DOMNodeInserted works smoothly in the admin.

**Comment by nameisnot** [comment]
@jpic Thanks for your reply. 

> did you try importing the admin js and way of doing ?

**1.** Please forgive me but could you explain how do I import admin js?
On the "**debug toolbar**" I can see `admin/js/vendor/jquery/jquery.js` being referred to from project's _**static**_ folder like this `<project_name>/static/admin/js/vendor/jquery/jquery.js`.

Under tab "Sources" in Developers Tools, I find the DAL tree like this:

![TREE](https://user-images.githubusercontent.com/40018225/65479846-bbeb1e00-deac-11e9-9915-8b298e2165d1.png)


**OR** is there something else that I need to do.

**2.** How and where do I check **_DOMNodeInserted_** event? Should it be done in "Developer Tool" ? I tried but can't find any reference. (BTW I am using Chrome).

In the meantime I will try and dig up some information on this. Whenever you get time to respond please do so, I will be delighted.

**Comment by jpic** [comment]
just did the example in commit e0b4aa7 which shows JS inlines outside the admin
![2019-09-24-165550_484x584_scrot](https://user-images.githubusercontent.com/94636/65523369-7c442680-deec-11e9-881a-327491985fdc.png)

**Comment by jpic** [comment]
@nameisnot is it working for you ?

**Comment by nameisnot** [comment]
@jpic Yes, but only so much as I explain below (forgive me for the voluminous reply):

**CASE 1:** 
I created a minimal sample of the app and getting exactly what you showed (below):

![ZDAL MINIMAL SAMPLE2809](https://user-images.githubusercontent.com/40018225/65816574-f50fef00-e21a-11e9-933a-c006ac5beb43.png)

**CASE 2:**
Whereas if I were to use the code I was using earlier, I was getting this:
![ZDAL MINIMAL ORIG 2809](https://user-images.githubusercontent.com/40018225/65816677-89c71c80-e21c-11e9-9673-08fad4c88a1a.png)

I have a feeling that my **_template_** may be at the root cause of the problem (but then again, I may be wrong!!).

**CASE 3:**
Now I tried to modify the template on which I am getting the result shown in **CASE 1**. The result is as shown below:

![ZDAL MIN MODIF DEBUG 2809](https://user-images.githubusercontent.com/40018225/65816769-6e104600-e21d-11e9-9ba0-963da448f390.png)

As you would have noticed, **ONLY** the first line of fields are **within the table** (I am trying to include them into), but the rest of the **added fields have their "label"** also and are thus not falling within the boundaries of the table.

Here is the template where I am iterating thru' the form fields:

```
{% extends "base.html" %}
{% load static %}

{% block content %}

<h1>New Purchase Orders Dynamic</h1>
<p><u>Create Purchase Orders</u></p>
<br/>
<div class="col-md-4">

<head>
<link rel="stylesheet" href="//code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css">

<style>
#resizable { width: 250px; height:250px; padding:0.8em; }
#resizable h3 { text-align: center; margin: 0; }
</style>

	<script src="{% static 'jquery.js' %}"></script>
	<script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>

<!-- FOLL IS FOR DYNAMICALLY ADDING FORMS -->
<script type="text/html" id="item-template">
<div id="item-__prefix__">
	{{ fresh_formset.empty_form }}
</div>
</script>

<!-- SCRIPT FOR ADDING FORMS -->
<script type="text/javascript"> 
$(document).ready(function() { 
    $('.add-item').click(function(ev) { 
        ev.preventDefault(); 
        var count = $('#items-form-container').children().length; 
        var tmplMarkup = $('#item-template').html(); 
        var compiledTmpl = tmplMarkup.replace(/__prefix__/g, count); 
        $('div#items-form-container').append(compiledTmpl); 

        // update form count 
        $('#id_item_items-TOTAL_FORMS').attr('value', count+1); 

        // some animate to scroll to view our new form 
        $('html, body').animate({ 
                scrollTop: $("#add-item-button").position().top-200 
            }, 800); 
    }); 
 }); 
 </script> 

<!-- END OF SCRIPT FOR ADDING FORMSET FORMS -->

<!-- THE FOLL STYLEs FOR UI -->
<style>

.table-wrapper-scroll-y {
  display: block;
  max-height: 400px;
  min-width:750px;
  max-width:750px;
  overflow-y: auto; 
  -ms-overflow-style: -ms-autohiding-scrollbar; 
{#  overflow-x: visible!important; #}
}

#resizable { width: 250px; height:250px; padding:0.8em; }
#resizable h3 { text-align: center; margin: 0; }

.tablehdr-wrapper-scroll-y {
  display: block;
  max-height: 400px;
  min-width:550px;
  max-width:550px;
  overflow-y: auto;
  -ms-overflow-style: -ms-autohiding-scrollbar;
  overflow: visible!important;
}

#poheader {
	border-collapse: collapse;
	width: 100%;
}

#poheader td, #poheader th {
	border: 1px solid #ddd;
	padding: 4px;
}

#poheader tr:nth-child(even){background-color: powderblue;}

#poheader tr:hover {background-color: #ddd;}

#poheader th {
	padding-top: 1px;
	padding-bottom: 1px;
	text-align: left;
	background-color: #279bbe;
	border-color: #1d7995;
	color: white;
	no-wrap: True;
}

#polinehdr tr {
	background-color: #279bbe;
	border-color: #1d7995;
{#	color: white; #}
}

#datetimepicker1 td {
	overflow: visible!important;
	z-index: 99999;
}

.td {
  overflow: visible!important;
}

#polinehdr tr:hover {background-color: #ddd;}
</style>

<style>
#resizable { width: 250px; height:250px; padding:0.8em; }
#resizable h3 { text-align: center; margin: 0; }
</style>
</head>
<!-- END OF HEAD SECTION. -->

<form action="" method="POST" class="form-inline">
{% csrf_token %}

<!-- BELOW IS THE HEADER FORM -->
<div class="col-md-4 tablehdr-wrapper-scroll-y">
<table id="poheader">
	{% for fieldY in new_poform %}
		<tr>
		<th>{{ fieldY.label_tag }}</th>
			<td>
				{{ fieldY.errors }}
				{{ fieldY }}
				{% if fieldY.help_text %}
					{{ fieldY.help_text|safe }}
				{% endif %}
			</td>
		</tr>
	{% endfor %}
</table>
</div>
<br>

<!-- BELOW IS THE START OF CHILD FORMS -->
	<div id="items-form-container" class="col-md-4 table-wrapper-scroll-y" style="font-family:'Courier New'">
		<table class="table table-bordered table-striped table-sm">
			{{ fresh_formset.management_form }}
			{% for form in fresh_formset.forms %}
				{% if forloop.first %}
					<thead id="polinehdr">
					<tr>
						{% for field in form.visible_fields %}
							<th>{{ field.label|capfirst }}</th>
						{% endfor %}
					</tr>
					</thead>
				{% endif %}
				{% for field in form.visible_fields %}
					<td>
						{# Include the hidden fields in the form #}
						{% if forloop.first %}
							{% for hidden in form.hidden_fields %}
								{{ hidden }}
							{% endfor %}
						{% endif %}
						{{ field.errors.as_ul }}
						{{ field }}
					</td>
				{% endfor %}
			{% endfor %}
		</table>
	</div>
<a href="#" id="add-item-button" class="btn btn-info add-item">Add Item Line</a>

	<input type="submit" class="btn btn-info" value="Save"/> 
	<a href="#" class="btn btn-info" role="button">Pricing</a> {# {% url 'purchorder_list' %} #}
	<a href="{% url 'purchorder_list' %}" class="btn btn-info" role="button">Cancel</a>

<br>
</form>
{% endblock %}

{% block footer %}
	<script type="text/javascript" src="{% static 'admin/js/vendor/jquery/jquery.js' %}"></script>
	{{ fresh_formset.media }}
{% endblock %}

```

Is there a way I can get rid of the labels from the added forms (and bring them inline with the first row)?

In case I am able to do that my life is made (at least for now).

Thanks so much.

**Comment by dwasyl** [comment]
@nameisnot the issue I was having on #1079 maybe be similar, the lines at issue are here:
[https://github.com/yourlabs/django-autocomplete-light/blob/master/src/dal/static/autocomplete_light/autocomplete.init.js#L93-L95](https://github.com/yourlabs/django-autocomplete-light/blob/master/src/dal/static/autocomplete_light/autocomplete.init.js#L93-L95)

Specifically:
`            $(e.target).find('[data-autocomplete-light-function=select2]:not([id*="__prefix__"])').each(initialize);`
needs to become:
`            $(e.target).find('[data-autocomplete-light-function=select2]').each(initialize);`

When that is changed in the `autocomplete.init.js` file, the jquery.formset.js plugin seems to work without any issue.

**Comment by dwasyl** [comment]
@jpic I ran into the same errors that @nameisnot ran into. I downgraded back to 3.3.5 and have had no issues at all (even on Django 2.2.6).

The issue that seems to keep coming up  seems to be with inline formsets on non-admin pages. There's something about the current version of the `autocomplete.init.js` that causes problem with initializing after being cloned.

**Comment by drrcodebase** [comment]
Hi @nameisnot ,  has the issue been resolved ? Please let me know I got the same issue.

**Comment by joshjung** [comment]
Same issue here as well. Please provide an update this is breaking our use of the Admin.

**Comment by drrcodebase** [comment]
as @dwasyl  mentioned , I have downgraded  to 3.3.5 version and which is working well. Thanks @dwasyl .  I am not sure why is it not working for 3.5.1

**Comment by jpic** [comment]
@drrcodebase this is supposed to be fixed in master and 3.7.0.dev0

Currently releasing 3.7.0
